container/new_pull.bzl

```diff
@@ -0,0 +1,208 @@
+
+This wraps the rulesdocker.go.cmd.puller.puller executable in a
+Bazel rule for downloading base images without a Docker client to
+construct new images.
+"""
+
+_container_pull_attrs = {
+    "architecture": attr.string(
+        default = "amd64",
+        doc = "(optional) Which CPU architecture to pull if this image " +
+              "refers to a multi-platform manifest list, default 'amd64'.",
+    ),
+    "cpu_variant": attr.string(
+        doc = "Which CPU variant to pull if this image refers to a " +
+              "multi-platform manifest list.",
+    ),
+    "digest": attr.string(
+        doc = "(optional: currently not supported) The digest of the image to pull.",
+    ),
+    "docker_client_config": attr.string(
+        doc = "A custom directory for the docker client config.json. " +
+              "If DOCKER_CONFIG is not specified, the value of the " +
+              "DOCKER_CONFIG environment variable will be used. DOCKER_CONFIG" +
+              " is not defined, the home directory will be used.",
+        mandatory = False,
+    ),
+    "os": attr.string(
+        default = "linux",
+        doc = "(optional) Which os to pull if this image refers to a " +
+              "multi-platform manifest list, default 'linux'.",
+    ),
+    "os_features": attr.string_list(
+        doc = "(optional) Specifies os features when pulling a multi-platform " +
+              "manifest list.",
+    ),
+    "os_version": attr.string(
+        doc = "(optional) Which os version to pull if this image refers to a " +
+              "multi-platform manifest list.",
+    ),
+    "platform_features": attr.string_list(
+        doc = "(optional) Specifies platform features when pulling a " +
+              "multi-platform manifest list.",
+    ),
+    "registry": attr.string(
+        mandatory = True,
+        doc = "The registry from which we are pulling.",
+    ),
+    "repository": attr.string(
+        mandatory = True,
+        doc = "The name of the image.",
+    ),
+    "tag": attr.string(
+        default = "latest",
+        doc = "(optional: currently unsupported) The tag of the image, default to 'latest' " +
+              "if this and 'digest' remain unspecified.",
+    ),
+    "_puller": attr.label(
+        executable = True,
+        default = Label("@go_puller//file:downloaded"),
+        cfg = "host",
+    ),
+}
+
+def _impl(repository_ctx):
+    """Core implementation of container_pull."""
+
+    # Add an empty top-level BUILD file.
+    repository_ctx.file("BUILD", "")
+
+    # TODO(xwinxu): change container_import rule to have an oci attribute to account for new OCI Image Layout
+    # currently exclude:     exports_files(["image.digest", "digest"]) in BUILD, theoretical BUILD commented out below:
+    repository_ctx.file("image/BUILD", """
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_docker//container:import.bzl", "container_import")
+
+container_import(
```

Comment:
Does this work? If it's broken.

I would probably make it a filegroup target like this
```
filegroup(
    name = "image",
    srcs = glob([
        "image/**",
    ]),
)
```

So the files can be referred.

Reply:
What I currently have works (at least no errors when I called the rule, and it pulled correctly). But I can try the filegroup as well